pkg/querier/queryrange/roundtrip.go

```diff
@@ -290,52 +290,66 @@ func NewLogFilterTripperware(
 	c cache.Cache,
 	metrics *Metrics,
 ) (queryrangebase.Tripperware, error) {
-	queryRangeMiddleware := []queryrangebase.Middleware{
-		StatsCollectorMiddleware(),
-		NewLimitsMiddleware(limits),
-		queryrangebase.InstrumentMiddleware("split_by_interval", metrics.InstrumentMiddlewareMetrics),
-		SplitByIntervalMiddleware(schema.Configs, limits, codec, splitByTime, metrics.SplitByMetrics),
-	}
+	return func(next http.RoundTripper) http.RoundTripper {
+		skipMiddleware := queryrangebase.NewRoundTripperHandler(next, codec)
```

Comment:
You shouldn't need to build this logic here. Each middleware is inherently passed a `next` `Handler` which can be called within the implementation.

For instance, `queryrangebase.Middleware` is the interface each middleware.

```go
// Middleware is a higher order Handler.
type Middleware interface {
	Wrap(Handler) Handler
}
```

It's common to use an adapter such as `queryrangebase.MiddlewareFunc`. Here's the example from `NewQueryShardMiddleware`:
```go
return queryrangebase.MiddlewareFunc(func(next queryrangebase.Handler) queryrangebase.Handler {
		return &shardSplitter{
			limits: limits,
			shardingware: queryrangebase.MergeMiddlewares(
				queryrangebase.InstrumentMiddleware("shardingware", middlewareMetrics),
				mapperware,
			).Wrap(next),
			now:  time.Now,
			next: queryrangebase.InstrumentMiddleware("sharding-bypass", middlewareMetrics).Wrap(next),
		}
	})
}
```

The `next` argument in this function is the following middleware handler, so the implementation can be roughly simplified to

```go
func(next queryrangebase.Handler) {
	if index.get(req) > max {
		return err
	}
	return next.do(req)
}
```

Reply:
edit: i'm wrong, resolving